osf/management/commands/create_EGAP_json.py

```diff
@@ -0,0 +1,251 @@
+import logging
+import csv
+import datetime
+import json
+import os
+import shutil
+import re
+
+from framework.celery_tasks import app as celery_app
+
+from django.core.management.base import BaseCommand
+from django.core.exceptions import ValidationError, ObjectDoesNotExist
+
+from osf.models import RegistrationSchema
```

Comment:
Can the schema be validated using the OSF as a library rather than running the OSF whole? It would be super-cool if we could break this into a script that we could put into an iPython notebook. Note that I'm not asking if we can re-write the schema validation, just whether the validation could be called without running the whole OSF. Does that make sense?

Reply:
I did a little looking, and it's not *so* bad. If we ditch the live link to the schema (and schemas don't change that much), we could just use https://github.com/CenterForOpenScience/osf.io/blob/develop/website/project/metadata/utils.py#L56 to convert the metaschema to json schema, save that in the script, then just use `jsonschemas` validators to do the validation. It's a little more work than I'd like to do, but it does keep us from having to make this an OSF.io PR. Is it worth it? I'm not sure.